unitycatalog/functions/client.py

```diff
@@ -0,0 +1,136 @@
+                of type InputParams in unitycatalog, or FunctionParameterInfos in databricks.
+            parameters (Dict[str, Any]): The parameters to validate.
+        """
+        parameters = deepcopy(parameters)
+        if input_params and input_params.parameters:
+            invalid_params: Dict[str, str] = {}
+            for param in input_params.parameters:
+                if param.name in parameters:
+                    try:
+                        self._validate_param_type(parameters.pop(param.name), param)
+                    except ValueError as e:
+                        invalid_params[param.name] = str(e)
+                elif param.parameter_default is None:
+                    raise ValueError(f"Parameter {param.name} is required but not provided.")
+            if invalid_params:
+                raise ValueError(f"Invalid parameters provided: {invalid_params}.")
+            if parameters:
+                raise ValueError(
+                    f"Extra parameters provided that are not defined in the function's input parameters: {parameters}."
+                )
+        elif parameters:
+            raise ValueError(
+                f"Function does not have input parameters, but parameters {parameters} were provided."
+            )
+
+    @abstractmethod
+    def _validate_param_type(self, value: Any, param_info: Any) -> None:
+        """
+        Validate the type of a parameter against the function's input parameter info.
+
+        Args:
+            value (Any): The value of the parameter.
+            param_info (Any): The parameter info.
+        """
+        raise NotImplementedError
+
+    def execute_function(
+        self, function_name: str, parameters: Optional[Dict[str, Any]] = None, **kwargs: Any
+    ) -> FunctionExecutionResult:
+        """
+        Execute a UC function by name with the given parameters.
+
+        Args:
+            function_name (str): The name of the function to execute.
+            parameters (Dict[str, Any], optional): The parameters to pass to the function. Defaults to None.
+            kwargs: additional key-value pairs to include when executing the function.
+
+        Returns:
+            FunctionExecutionResult: The result of executing the function.
+        """
+        with self._lock:
+            function_info = self.retrieve_function(function_name, **kwargs)
+            parameters = parameters or {}
+            self.validate_input_params(function_info.input_params, parameters)
+            return self._execute_uc_function(function_info, parameters, **kwargs)
+
+    @abstractmethod
+    def _execute_uc_function(
+        self, function_info: Any, parameters: Dict[str, Any], **kwargs: Any
+    ) -> FunctionExecutionResult:
+        raise NotImplementedError
+
+
+# TODO: update BaseFunctionClient to Union[BaseFunctionClient, AsyncBaseFunctionClient] after async client is supported
+def get_uc_function_client() -> Optional[BaseFunctionClient]:
```

Comment:
how is this function used?

Reply:
It's not used yet, it will be used in langchainToolkit, something like below when defining the tool's func:
```
def func(*args: Any, **kwargs: Any) -> str:
        client = get_uc_function_client()
        args_json = json.loads(json.dumps(kwargs, default=str))
        result = client.execute_function(
            function_name=func_name,
            parameters=args_json,
        )
        return result.to_json()
```